src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -52,7 +63,7 @@ class Crystal::Doc::Generator
       Markdown.parse body, MarkdownDocRenderer.new(program_type, io)
     end
 
-    File.write "#{@dir}/index.html", MainTemplate.new(body, types, repository_name)
+    File.write "#{@dir}/index.html", MainTemplate.new(generate_flags(body), types, repository_name)
```

Comment:
Fixed it. I even made an optimization where it won't even run the markdown parser on an empty string for the README if the file doesn't exist.